### ISSUE TITLE
Develop a massa-proto TS/JS

### ISSUE THREAD

**Issue by loonggamma** [massalabs/massa-proto-as#14]
**The need:**
Products with TS/JS stack (such as Explorer, Massaexplo) would benefit from a JS/TS source code from protobuf files.

**Comment by modship** [comment]
> **The need:** Products with TS/JS stack (such as Explorer, Massaexplo) would benefit from a JS/TS source code from protobuf files.

hi @longgamma-massa , can u explain me more what do you need ?

**Comment by bilboquet** [comment]
@longgamma-massa 
I guess this is to avoid use of Args.
Am I correct if I point this repo https://github.com/massalabs/massa-proto-as (where we have done some similar job for our second ABI) and say this is the kind of things you want to generalize?

EDIT:
More precisely, I'm talking about https://github.com/massalabs/as_abi_protobuf/ sorry the repo is private to Massalabs for now :(

**Comment by gregLibert** [comment]
@bilboquet @modship No, we just want to simplify the work for people wanting to interact with the node or the indexer  API.
I guess what we are looking for here is a TypeScript-specific version of [massa-proto](https://github.com/massalabs/massa-proto) to deserialize protobuf binary content.

**Comment by bilboquet** [comment]
I was just mentioning those repo as examples because my understanding is that they respond to the same need and I fell that [massa-proto-as](https://github.com/massalabs/massa-proto-as) is a good starting point to create a `massa-proto-js`
edit: at least a POC
